### DISCUSSION TITLE
The Eclipse Jetty version 9.4.x isn't compatible with the jarkata.servlet-api 4.0.4

### DISCUSSION THREAD

**Discussion by jokyjang** [dropwizard/dropwizard#6079] [General]
Looks like the new servlet-api version has new method in the interface ServletContext which failed the jetty

```
        javax.servlet.ServletException: Unable to find required instance of org.eclipse.jetty.websocket.server.NativeWebSocketConfiguration at ServletContext attribute 'org.eclipse.jetty.websocket.server.NativeWebSocketConfiguration'
```

**Comment by joschi** [comment]
@jokyjang That's correct, Jetty 9.4.x is targeting an older version of the Java EE Servlet API.

Jetty and by extension Dropwizard 2.x will not support newer versions of the Servlet API.

If you want or need to use a later version of the Java EE or Jakarta EE Servlet API, please take a look at Dropwizard 3.x (Java EE) or 4.x (Jakarta EE) respectively.

**Reply by zUniQueX** [reply]
@joschi IIRC we've bumped the servlet API to 4.x in the Dropwizard 2.1.x baseline some time ago, although Jetty 9.4.x isn't compatible with it. We've already discussed reverting the upgrade but decided on leaving this as-is.

**Reply by jokyjang** [reply]
Thanks @joschi and @zUniQueX for the explain. I'll see if we can upgrade to 3.x

**Reply by zUniQueX** [reply]
Please note, that DW 3.x is currently in beta stage. We may introduce further breaking changes (and have planned to do so), until the final version gets released.
